From c5c483f5eee99dbb39a1749767fdd649ed640211 Mon Sep 17 00:00:00 2001
From: Anurag Kumar Vulisha <anurag.kumar.vulisha@xilinx.com>
Date: Thu, 23 Jan 2020 19:51:11 +0530
Subject: [PATCH 1021/1852] usb: host: xhci: Add xhci_hcd to xhci_ring for use
 in stream_timeout_function

commit 59f610add3418994aaace29508a81ec55d90a5c9 from
https://github.com/Xilinx/linux-xlnx.git

This patch modifies the code to pass xhci_hcd in xhci_ring
which would be used by stream_timeout_function() to send
stop command on the stream ring after stream timer gets
timedout.

Signed-off-by: Anurag Kumar Vulisha <anurag.kumar.vulisha@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Piyush Mehta <piyush.mehta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/usb/host/xhci-mem.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/host/xhci-mem.c b/drivers/usb/host/xhci-mem.c
index 0e290b7dfd62..456f5d0667fe 100644
--- a/drivers/usb/host/xhci-mem.c
+++ b/drivers/usb/host/xhci-mem.c
@@ -692,6 +692,7 @@ struct xhci_stream_info *xhci_alloc_stream_info(struct xhci_hcd *xhci,
 			 */
 			timer_setup(&cur_ring->stream_timer,
 				xhci_stream_timeout, 0);
+			cur_ring->xhci = xhci;
 		}
 
 		ret = xhci_update_stream_mapping(cur_ring, mem_flags);
-- 
2.31.1

